docs/ALU Design.md

```diff
@@ -0,0 +1,801 @@
+# ALU design
+
+This document describes the research done for
+[#27 Design ALU](https://github.com/reilabs/llvm-to-cairo/issues/27).
+
+The first part, **Research**, describes selected features of LLVM IR, Rust and Cairo (both the
+virtual machine and the programming language), that impact the way we must handle arithmetic and
+logic operations. The second part, **Design**, specifies decisions made with regard to the shape of
```

Comment:
You can link "research" and "design" to the corresponding internal headers here. It'll help with document navigation.